cli/src/main/kotlin/com/wire/kalium/cli/SendReceiveApplication.kt

```diff
@@ -0,0 +1,89 @@
+package com.wire.kalium.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.configuration.ServerConfig
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.feature.auth.AuthSession
+import com.wire.kalium.logic.feature.auth.AuthenticationResult
+import com.wire.kalium.logic.feature.client.RegisterClientResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+class SendReceiveApplication : CliktCommand() {
+
+    private val email: String by option(help = "Wire account email").required()
+    private val password: String by option(help = "Wire account password").required()
+    private val environment: String? by option(help = "Choose backend environment: can be production or staging")
+    private val coreLogic = CoreLogic("Kalium CLI", ".proteus")
+
+    private val serverConfig: ServerConfig by lazy {
+        if (environment == "production") {
+            ServerConfig.PRODUCTION
+        } else {
+            ServerConfig.DEFAULT
+        }
+    }
+
+    override fun run() = runBlocking {
+        val authSession = login(email, password)
+        val userSession = coreLogic.getSessionScope(authSession)
+
+        when (userSession.client.register(password, emptyList())) {
+            is RegisterClientResult.Failure -> throw RuntimeException("Client registration failed")
+            is RegisterClientResult.Success -> Unit
+        }
+
+        val conversations = userSession.conversations.getConversations().first()
+
+        conversations.forEachIndexed { index, conversation ->
+            println("$index) ${conversation.id.value}  Name: ${conversation.name}")
+        }
+
+        print("Enter conversation index: ")
+        val conversationIndex = readLine()!!.toInt()
+        val conversationID = conversations.get(conversationIndex).id
+
+        launch(Dispatchers.Default) {
+            userSession.listenToEvents()
+        }
+
+        launch(Dispatchers.Default) {
+            userSession.messages.getRecentMessages(conversationID, limit = 1).collect {
+                for (message in it) {
+                    val content = message.content
+
+                    when (content) {
+                        is MessageContent.Text -> println("> ${content.value}")
+                        else -> println("<unknown message>")
```

Comment:
Do we need to show `<unknown message>` since this will happen for many messages e.g., message read, typing etc …?
```suggestion
                        Unknown -> { /* do nothing */}
```

Reply:
No I guess we don't really need to show it.